packages/picker/src/MobileController.ts

```diff
@@ -28,6 +29,16 @@ export class MobileController extends InteractionController {
         this.preventNextToggle = this.open ? 'yes' : 'no';
     }
 
+    public override handleButtonFocus(event: FocusEvent): void {
+        super.handleButtonFocus(event);
+
+        // if the focus comes from a closing the menu, we need to make sure to
+        // unset the :focus-visible state to avoid the focus ring on safari mobile
+        if (isWebKit() && !this.open) {
+            this.target.blur();
```

Comment:
Are we sure that `.blur()` is the desired approach here? From my understanding, using .blur() will remove focus from the element entirely, but our intention seems to be to keep the element focused while only removing the visual indication (the focus ring). Or how does it work in the other browsers?
Can we check where the focus goes with this implementation? Does it impact keyboard navigation?